lib/embulk/guess/schema_guess.rb

```diff
@@ -62,11 +62,8 @@ def guess_type(str)
           return "long"
         end
 
-        if str.include?('.')
-          a, b = str.split(".", 2)
-          if a.to_i.to_s == a && b.to_i.to_s == b
-            return "double"
-          end
+        if str =~ /^[-+]?([1-9]\d*|0)(\.\d+)([eE][-+]?[0-9]+)?$/
```

Comment:
1. Can you leave a comment, the core intention of this regular expression?
2. Do you have any reason why it has both `\d` and `[0-9]`?
3. Can we have a help from some external libraries, or a kind of Ruby's or Java's standard parsers?
4. Do we have any explicit spec what strings are guessed as `double`? (I don't think so because it's just a guess, though.)

I was not sure what behavior is expected, for example:
* `0.0`
* `00.1`
* ` 12.0000`
* `1e+2`
* `1e-2`